src/service-implementations/lf-repo-tree-node-service.ts

```diff
@@ -0,0 +1,371 @@
+   *            - FolderInFolderInRoot
+   *            - FolderInFolderInRoot1
+   *            - FolderInFolderInRoot2
+   *            ...
+   *            - FolderInFolderInRoot17
+   *            - FolderInFolderInRoot18
+   *            - FolderInFolderInRoot19
+   *            - FolderInFolderInRoot20
+   *            - FolderInFolderInRoot21
+   *      - FolderInRoot1
+   *      - FolderInRoot2
+   *      - FolderInRoot3
+   * const service = new LfRepoTreeNodeService(repoClient);
+   * service.getRootTreeNodeAsync(FolderInRoot) -> root
+   * ```
+   */
+  async getRootTreeNodeAsync(): Promise<LfRepoTreeNode> {
+    const repoName: string = await this.repoClient.getCurrentRepoName();
+    const repoId: string = await this.repoClient.getCurrentRepoId();
+    const rootFolderId: number = 1;
+    const rootEntry: Entry = await this.repoClient.entriesClient.getEntry(
+      {
+        repoId,
+        entryId: rootFolderId
+      }
+    );
+    return this.createNode(rootEntry, repoName);
+  }
+
+  private isViewable(entry: Entry): boolean {
+    return entry.entryType !== EntryType.RecordSeries && (this.viewableEntryTypes.includes(entry.entryType)
+      || (entry.entryType === EntryType.Shortcut && this.viewableEntryTypes.includes((entry as Shortcut).targetType)));
+  }
+
+  private createNode(entry: Entry, repoName: string, parent?: LfRepoTreeNode): LfRepoTreeNode {
+    let treeNode: LfRepoTreeNode | undefined;
+    if (!entry.entryType) {
+      throw new Error('entry type is undefined');
+    }
+    if (entry.entryType === EntryType.Folder) {
+      treeNode = this.createFolderNode(entry, repoName, parent);
+    } else if (entry.entryType === EntryType.Document) {
+      treeNode = this.createLeafNode(entry, parent);
+    } else if (entry.entryType === EntryType.Shortcut) {
+      const shortcut = entry as Shortcut;
+      if (shortcut.targetType === EntryType.Folder) {
+        treeNode = this.createFolderNode(shortcut, repoName, parent);
+      } else if (shortcut.targetType === EntryType.Document) {
+        treeNode = this.createLeafNode(shortcut, parent);
+      }
+      else {
+        throw new Error('Unexpected shortcut targetType');
+      }
+    } else {
+      throw new Error('Unsupported entry type');
+    }
+    this.setNodeAttributes(treeNode, entry);
+    return treeNode;
+  }
+
+  private createLeafNode(entry: Entry, parent?: LfRepoTreeNode): LfRepoTreeNode {
+
+    const path = parent ? PathUtils.combinePaths(parent.path, entry.name) : entry.fullPath;
+    if (!path) {
+      throw new Error('entry fullPath is undefined');
+    }
+
+    const leafNode: LfRepoTreeNode = {
+      name: entry.name!,
+      path: path,
+      id: entry.id!.toString(),
+      entryType: entry.entryType,
+      icon: [],
+      isContainer: false,
+      isLeaf: true,
+      attributes: new Map<string, any>()
+    };
+    return leafNode;
+  }
+
+  private setNodeAttributes(node: LfRepoTreeNode, entry: Entry): void {
```

Comment:
I think this function should be named a bit different, it doesn't just set attributes, but also adds all the shortcut specific stuff, and updates the icons

maybe setNodeProperties or updateNodeSpecificProperties? -- I just don't like attributes bc it's the name of one member of node